services/mgmt/compute/examples/vm_list_using_resource_graph.rs

```diff
@@ -0,0 +1,88 @@
+use std::sync::Arc;
+use azure_identity::DefaultAzureCredential;
+use azure_mgmt_resourcegraph::models::{QueryRequest, QueryRequestOptions};
+use serde_json::{json, Value};
+use azure_mgmt_resourcegraph::Client as AzureResourceGraphClient;
+use azure_mgmt_resourcegraph::ClientBuilder as AzureResourceGraphClientBuilder;
+
+/// Fetching Virtual Machines using Azure Resource Graph
+
+#[derive(Debug)]
+pub enum GenericError {
+    InvalidRequest,
+}
+
+#[derive(Debug)]
+struct CustomError {
+    err_type: GenericError,
+    err_msg: String
+}
+
+async fn fetch_azure_records() -> Result<Value, CustomError> {
+
+    let azure_creds = Arc::new(DefaultAzureCredential::default());
+
+    let az_rg_client = AzureResourceGraphClient::builder(azure_creds).build();
+
+    // fetch first 10 records
+    let query_options = QueryRequestOptions {
+        skip_token: None,
+        top: Option::from(10),
+        skip: Option::from(0),
+        result_format: None,
+        allow_partial_scopes: None,
+        authorization_scope_filter: None,
+    };
```

Comment:
QueryRequestOptions implements [std::default::Default](https://doc.rust-lang.org/std/default/trait.Default.html).

 As such, this could be simplified via:
```suggestion
    let query_options = QueryRequestOptions {
        top: Option::from(10),
        skip: Option::from(0),
        ..Default::default()
    };
```

Reply:
Yep, that looks better :)